examples/hybrid/staggered_nonhydrostatic_model.jl

```diff
@@ -328,9 +328,15 @@ function remaining_tendency!(Yₜ, Y, p, t)
         (; enable_default_remaining_tendency) = p
         Yₜ .= zero(eltype(Yₜ))
         if enable_default_remaining_tendency
-            default_remaining_tendency!(Yₜ, Y, p, t)
+            if :ρe_tot in propertynames(Y.c)
+                default_remaining_tendency_special!(Yₜ, Y, p, t)
+            else
+                default_remaining_tendency_generic!(Yₜ, Y, p, t)
+            end
```

Comment:
We should probably add `default_remaining_tendency!` to the cache, and determine which is used in the driver. One issue that I see is that `default_remaining_tendency_special!` is effectively the default and will be used for both the threaded and unthreaded cases, which seems odd. Should be an easy thing to change